includes/wc-product-functions.php

```diff
@@ -105,7 +105,11 @@ function wc_get_product( $the_product = false, $deprecated = array() ) {
 	if ( ! empty( $deprecated ) ) {
 		wc_deprecated_argument( 'args', '3.0', 'Passing args to wc_get_product is deprecated. If you need to force a type, construct the product class directly.' );
 	}
-	return WC()->product_factory->get_product( $the_product, $deprecated );
+
+	$product = WC()->product_factory->get_product( $the_product, $deprecated );
+
+	// Return null on failure for strict backwards compatibility.
+	return $product ? $product : null;
```

Comment:
Is this bw compat? Factory get_product returns bool on failure. Maybe the docblock just needs updating and this code removed.